improver_tests/utilities/test_compare.py

```diff
@@ -129,7 +192,7 @@ def message_collector(message):
         "root", actual_ds, expected_ds, message_collector)
     assert len(messages_reported) == 0
 
-    # Check modifying an attribute
+    # Check modifying a simple attribute
```

Comment:
I assume this has been kept together as one enormous test as all the changes are related to attributes? I'd prefer it to be broken up into tests that covered the various elements in isolation, but that's a stylistic thing; the functionality looks fine.